R/prepare-lpr.R

```diff
@@ -0,0 +1,115 @@
+#' Prepare the two LPR2 registers into one to extract diabetes diagnoses data.
+#'
+#' The output is used as inputs to `include_diabetes_diagnoses()` and to
+#' `get_pregnancy_dates()` (see exclusion events).
+#'
+#' @param lpr_diag The LPR2 register containing diabetes diagnoses.
+#' @param lpr_adm The LPR2 register containing hospital admissions.
+#'
+#' @return The same type as the input data, default as a [tibble::tibble()],
+#'  with the following columns:
+#'
+#'  -   `pnr`: The personal identification variable.
+#'  -   `date`: The date of all the recorded diagnosis (renamed from `d_inddto`).
+#'  -   `is_primary_diagnosis`: Whether the diagnosis was a primary diagnosis.
+#'  -   `has_diabetes`: Whether the diagnosis was any type of diabetes.
+#'  -   `has_t1d`: Whether the diagnosis was T1D-specific.
+#'  -   `has_t2d`: Whether the diagnosis was T2D-specific.
+#'  -   `has_pregnancy_event`: Whether the person has an event related to pregnancy like giving birth or having a miscarriage at the given date.
+#'  -   `is_endocrinology_department`: Whether the diagnosis was made made by an
+#'      endocrinology (TRUE) or other medical (FALSE) department.
+#'
+#' @keywords internal
+#' @inherit algorithm seealso
+#'
+#' @examples
+#' \dontrun{
+#' register_data <- simulate_registers(c("lpr_diag", "lpr_adm"), 100000)
+#' lpr2 <- prepare_lpr2(
+#'   lpr_diag = register_data$lpr_diag,
+#'   lpr_adm = register_data$lpr_adm
+#' )
+#' }
+prepare_lpr2 <- function(lpr_adm, lpr_diag) {
+  logic <- c(
+    "lpr2_needed_codes",
+    "lpr2_has_pregnancy_event",
+    "lpr2_has_t1d",
+    "lpr2_has_t2d",
+    "lpr2_has_diabetes",
+    "lpr2_is_endocrinology_department",
+    "lpr2_is_medical_department"
+  ) |>
+    rlang::set_names() |>
+    purrr::map(get_algorithm_logic) |>
+    # To convert the string into an R expression
+    purrr::map(rlang::parse_expr)
+
+  lpr_diag |>
+    column_names_to_lower() |>
+    dplyr::filter(!!logic$lpr2_needed_codes) |>
+    dplyr::inner_join(
+      column_names_to_lower(lpr_adm),
```

Comment:
```suggestion
     lpr_adm,
```
Bc that's done on L49?

Reply:
No, that is only done for `lpr_diag`, not `lpr_adm` :relaxed: